pilot/pkg/config/kube/gateway/model.go

```diff
@@ -83,7 +87,11 @@ func (refs AllowedReferences) BackendAllowed(
 ) bool {
 	from := Reference{Kind: k, Namespace: k8s.Namespace(routeNamespace)}
 	to := Reference{Kind: gvk.Service, Namespace: backendNamespace}
-	allow := refs[from][to]
+	mFromMap, ok := refs[from]
```

Comment:
why this change? it doesn't seem to be fixing any bugs or making the code simpler

Reply:
It may cause a NPE

Reply:
I don't think that is true, read of nil map is ok.

Reply:
yes, i reverted it, thanks for your review!